app/controllers/api_controller/providers.rb

```diff
@@ -132,13 +137,20 @@ def validate_credential_attributes(provider, creds)
                                     invalid_attrs.join(', '), provider.class.name, auth_attrs.join(", "))
     end
 
+    def check_endpoints(data)
+      return unless data["connection_configurations"]
+      raise BadRequestError, "Attribute connection_configurations must be an Array" unless
+        data["connection_configurations"].kind_of?(Array)
+    end
+
     def fetch_provider_data(provider_klass, data, options = {})
       provider_data = data.except(*RESTRICTED_ATTRS)
       invalid_keys  = provider_data.keys - provider_klass.columns_hash.keys - ENDPOINT_ATTRS
       raise BadRequestError, "Invalid Provider attributes #{invalid_keys.join(', ')} specified" if invalid_keys.present?
 
+      check_endpoints(data)
```

Comment:
It's not quite clear when this will apply.  Looking at callers of "fetch_provider_data", it will run on create & edit.

=> The existing method's name is just bad, sounds like something that happens on GET.
Consider renaming to e.g. "sanitize_provider_data".